common/interfaces/card_interface/nfc_events.c

```diff
@@ -178,6 +178,7 @@ void nfc_set_card_removed_event(void) {
 
 void nfc_en_select_card_task(void) {
   nfc_state = NFC_STATE_SET_SELECT_CARD_CMD;
+  nfc_deselect_card();
```

Comment:
why this addition is required?

Reply:
If an NFC event occurs and the card is not removed, and then waiting for the NFC event, card is not detected.

Steps to recreate issue:
- Call `card_initialize_applet`
- Tap card but don't lift
- Call `card_initialize_applet` again (hence the NFC event again)

Expected behaviour is that card will not be detected on the second call.


Deselecting the card before waiting for event again solves this.

Reply:
is there any relevant place to add this knowledge?

Reply:
I'll add a comment here with the change itself.
I haven't done the complete RCA on why the selection is not working on the second event. This might even be a temporary solution.

The idea for this change is to deselect the card just before selecting it, so we don't have to deselect in every flow.

Reply:
added